action/protocol/execution/evm/evmstatedbadapter.go

```diff
@@ -344,6 +345,9 @@ func (stateDB *StateDBAdapter) GetNonce(evmAddr common.Address) uint64 {
 		}
 		pendingNonce--
 	}
+	if pendingNonce == 1 && stateDB.convertFreshAddress {
```

Comment:
could we remove `pendingNonce == 1`? same question for all the other places making a similar check outside of function `AdjustNonceForFreshAccount`